eventsources/sources/github/validate.go

```diff
@@ -31,15 +31,13 @@ func validate(githubEventSource *v1alpha1.GithubEventSource) error {
 	if githubEventSource == nil {
 		return common.ErrNilEventSource
 	}
-	if githubEventSource.Repository == "" {
-		return fmt.Errorf("repository cannot be empty")
+	if githubEventSource.GetOwnedRepositories() == nil {
+		return fmt.Errorf("no valid repository owner and name found")
 	}
-	if githubEventSource.Owner == "" {
-		return fmt.Errorf("owner cannot be empty")
-	}
-	if githubEventSource.Events == nil || len(githubEventSource.Events) < 1 {
-		return fmt.Errorf("events must be defined")
+	if githubEventSource.APIToken != nil && githubEventSource.Webhook.URL != "" && len(githubEventSource.Events) == 0 {
```

Comment:
minor - you could add githubEventSource.IsValid

Reply:
Updated.